src/components/apps/launch/InputSelector.js

```diff
@@ -84,6 +88,19 @@ const InputSelector = ({ intl, acceptedType, startingPath, ...props }) => {
 
     const inputProps = {
         readOnly: true,
+        endAdornment:
+            <InputAdornment position="end">
+                <BrowseButton
+                    baseId={id}
+                    startingPath={startingPath}
+                    acceptedType={acceptedType}
+                    multiSelect={false}
+                    name={field.name}
+                    onConfirm={(selections) => {
+                        setFieldValue(field.name, selections);
+                    }}
+                />
+            </InputAdornment>
```

Comment:
This change made `InputSelector` as a single widget. That fixed the `Browse`  button mis alignment issue.

Reply:
For optional input fields, this will show `Browse` when the field is empty and when the field is populated, it will show `X` button.